tests/test_router_bgp.rb

```diff
@@ -524,47 +535,45 @@ def test_routerbgp_default_confederation_id
     bgp.destroy
   end
 
-  def test_routerbgp_set_get_confederation_peers
-    skip(XR_SUPPORTED_BROKEN) if platform == :ios_xr
-    %w(test_default test_vrf).each do |t|
-      if t == 'test_default'
-        asnum = 55
-        vrf = 'default'
-        bgp = RouterBgp.new(asnum)
-      else
-        asnum = 99
-        vrf = 'yamllll'
-        bgp = create_bgp_vrf(asnum, vrf)
-      end
-      # Confederation peer configuration requires that a
-      # confederation id be configured first so the expectation
-      # in the next test is an empty peer list
-      bgp.confederation_id = 55
-      assert_empty(bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be empty")
-      bgp.confederation_peers_set(15)
-      assert_equal('15', bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be '15'")
-      bgp.confederation_peers_set(16)
-      assert_equal('16', bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be '16'")
-      bgp.confederation_peers_set(55.77)
-      assert_equal('55.77', bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be" \
-                   "'55.77'")
-      bgp.confederation_peers_set('15 16 55.77 18 555 299')
-      assert_equal('15 16 55.77 18 555 299',
-                   bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be" \
-                   "'15 16 55.77 18 555 299'")
-      bgp.confederation_peers_set('')
-      assert_empty(bgp.confederation_peers,
-                   "vrf #{vrf}: bgp confederation_peers list should be empty")
-      bgp.destroy
-    end
+  def test_confederation_peers_default
+    confed_peers_test(setup_default)
+  end
+
+  def test_confederation_peers_vrf
+    skip(XR_NO_VRF_SUPPORT) if platform == :ios_xr
+    confed_peers_test(setup_vrf)
+  end
+
+  def confed_peers_test(bgp)
+    # Confederation peer configuration requires that a
+    # confederation id be configured first so the expectation
+    # in the next test is an empty peer list
+    bgp.confederation_id = 55
+
+    assert_empty(bgp.confederation_peers,
+                 'bgp confederation_peers list should be empty')
+    bgp.confederation_peers = [15]
+    assert_equal(['15'], bgp.confederation_peers,
+                 "bgp confederation_peers list should be ['15']")
+    bgp.confederation_peers = [16]
+    assert_equal(['16'], bgp.confederation_peers,
+                 "bgp confederation_peers list should be ['16']")
+    bgp.confederation_peers = [55.77]
+    assert_equal(['55.77'], bgp.confederation_peers,
+                 'bgp confederation_peers list should be ' \
+                 "['55.77']")
+    bgp.confederation_peers = ['15', '55.77', '16', '18', '555', '299']
+    assert_equal(['15', '16', '18', '299', '55.77', '555'],
```

Comment:
I wonder - is this the 'correct' sorting order, or should '55.77' come after '555' (since it's actually a 32-bit number). Don't know if this actually matters in any case, just wondering.